research/03-literate-prog.tex

```diff
@@ -1,46 +1,60 @@
+
+Even though the idea was conceived over 30 years ago, implementations are not
+very common. However, in recent years the idea seems to gain popularity again.
+A very recent implementation of literate programming is Apple's Swift
+playgrounds~\cite{swift-playgrounds}. Swift playgrounds are interactive
+documents in which code is executed and run as it is typed, in contrast to the
+non-interactive style of the first ``WEB'' language.
+
+In recent years, there has also been a particular focus on reproducible
+research. While literate programming primarily aims to add documentation to
+code, reproducible research focuses on adding code to documentation. More
+specifically, reproducible research refers to the idea that scientific papers
+should be augmented with the computer code used to carry out the
+research~\cite{schulte2012}. Examples of recent projects claiming to support
+both reproducible research and literate programming include the IPython
+project~\cite{ipython2007} and Emacs Org-mode~\cite{schulte2012}.
 
 \subsubsection{The ``WEB'' language}
 
-The ``WEB'' language was mainly a combination of two other languages, namely
-\TeX{} for document formatting and PASCAL as the programming language. By
-combining these languages, each with a different purpose, into one language, the
-source code serves two purposes: to produce a document describing the program
-and to produce a machine-executable program. Since both program and
+The original ``WEB'' language was mainly a combination of two other languages,
+namely \TeX{} for document formatting and PASCAL as the programming language.
+By combining these languages, each with a different purpose, into one language,
+the source code serves two purposes: to produce a document describing the
+program and to produce a machine-executable program. Since both program and
 documentation are generated from the same source file, they are always
 consistent with each other.
 
 \subsubsection{IPython / Jupyter notebooks}
 
 A more recent example of literate programming comes in the form of IPython with
 Jupyter notebooks~\cite{ipython2007}. IPython was partially inspired by other
-scientific tools already offering notebook-like functionality, such as Matlab or
-Mathematica. Besides the literal programming style introduced by ``WEB'', IPython
-notebooks allow for REPL-style interactive editing; documentation and code can
-be edited live and blocks of code can be reevaluated, printing their updated
-results. IPython notebooks also allow for more complex graphical elements such
-as 3D-plots. See \cref{fig:ipython} for an example of an IPython notebook.
+scientific tools already offering notebook-like functionality, such as Matlab
+or Mathematica. Since its inception, the project has been split off into the
+IPython, which provides an interactive REPL and a kernel for Jupyter, and
+Jupyter notebooks, which provide the notebook format and web application. Like
```

Comment:
So... It provides (or the notebooks provide?) the GUI. I think that's a better way to put it.